core/src/cas/content.rs

```diff
@@ -0,0 +1,86 @@
+use hash::HashString;
+use multihash::Hash;
+
+/// an Address for some Content
+/// ideally would be the Content but pragmatically must be HashString
+/// consider what would happen if we had multi GB addresses...
+pub type Address = HashString;
+/// the Content is a String
+/// this is the only way to be confident in persisting all Rust types across all backends
+pub type Content = String;
+
+/// can be stored as serialized content
+/// the content is the address, there is no "location" like a file system or URL
+/// @see https://en.wikipedia.org/wiki/Content-addressable_storage
+pub trait AddressableContent {
+    /// the Address the Content would be available at once stored in a ContentAddressableStorage
+    /// default implementation is provided as hashing Content with sha256
+    /// the default implementation should cover most use-cases
+    /// it is critical that there are no hash collisions across all stored AddressableContent
+    /// it is recommended to implement an "address space" prefix for address algorithms that don't
```

Comment:
@ddd-mtl does this satisfy your concerns?

i feel `address()` should satisfy the expectation that it be universally unique, or as close to as is practical

there's two ways to achieve this:

- use a hash
- use a unique prefix

Reply:
yes